src/data/basic-transform-mixin.js

```diff
@@ -0,0 +1,37 @@
+
```

Comment:
I've asked this a couple of times, but I'd still like to hear the rationale for transforming data into new fields. This seems complicated to me, but if it makes the data pipeline clearer in some way, I'd like to hear about it.

I will repeat my observation that there are literally dozens of accessors, and it seems messy to me to create new fields for them rather than just calling the accessors.

I guess as long as you are doing this every redraw pass, and there is a clear... what's the word I'm looking for? provenance? pipeline? data flow? then it could make sense. I just want to understand the rationale.

It does seem like chaining `.data()` from parent to child provides a clear flow, so maybe the risk of duplicated and inconsistent data is mitigated,